controllers/nnf_workflow_controller_helpers.go

```diff
@@ -198,36 +198,81 @@ func (r *NnfWorkflowReconciler) validateStagingDirective(ctx context.Context, wf
 func (r *NnfWorkflowReconciler) validateContainerDirective(ctx context.Context, wf *dwsv1alpha1.Workflow, directive string) error {
 	args, err := dwdparse.BuildArgsMap(directive)
 	if err != nil {
-		return nnfv1alpha1.NewWorkflowError("Invalid DW directive: " + directive).WithFatal()
+		return nnfv1alpha1.NewWorkflowError("invalid DW directive: " + directive).WithFatal()
+	}
+
+	// Ensure the supplied profile exists or use the default
+	profile, err := findContainerProfileToUse(ctx, r.Client, args)
+	if err != nil {
+		return nnfv1alpha1.NewWorkflowError(err.Error())
 	}
 
-	// Find the wildcard JOB_DW and PERSISTENT_DW arguments
-	for arg, val := range args {
-		// log.Log.Info("BLAKE: arg", "arg", arg)
+	// Check to see if the directive argument is in the list of storages in the container profile
+	checkDirectiveIsInProfile := func(storageName string) error {
```

Comment:
"Directive" here threw me off. I think of a directive as the whole #DW line. This seems to be checking whether a storage allocation is included in the profile.

Reply:
Good point. I will change to something like `checkStorageIsInProfile`.